op-interop-mon/monitor/metric_collector.go

```diff
@@ -20,14 +21,18 @@ type MetricCollector struct {
 	closed chan struct{}
 	log    log.Logger
 	m      InteropMessageMetrics
+
+	// Failsafe client for triggering failsafe enable/disable
+	failsafeClient FailsafeClient
 }
 
-func NewMetricCollector(log log.Logger, m InteropMessageMetrics, updaters map[eth.ChainID]Updater) *MetricCollector {
+func NewMetricCollector(log log.Logger, m InteropMessageMetrics, updaters map[eth.ChainID]Updater, failsafeClient FailsafeClient) *MetricCollector {
 	return &MetricCollector{
-		log:      log,
-		m:        m,
-		updaters: updaters,
-		closed:   make(chan struct{}),
+		log:            log,
+		m:              m,
+		updaters:       updaters,
+		failsafeClient: failsafeClient,
```

Comment:
Can you see any value in expanding the client a bit so that we can do some basic healthcheck on the supervisor when the monitor starts up? Otherwise, we won't detect a misconfiguration until a critical moment when it will be too late.